core/server/api_container/server/startosis_engine/plan_yaml/plan_yaml_generator.go

```diff
@@ -0,0 +1,502 @@
+
+	planYaml.addTaskYaml(taskYaml)
+	return nil
+}
+
+func (planYaml *PlanYaml) AddExec(
+	serviceName string,
+	returnValue *starlark.Dict,
+	cmdList []string,
+	acceptableCodes []int64) error {
+	uuid := planYaml.generateUuid()
+
+	// store future references
+	codeVal, found, err := returnValue.Get(starlark.String("code"))
+	if err != nil {
+		return err
+	}
+	if !found {
+		return stacktrace.NewError("No code value found on exec dict")
+	}
+	codeFutureRef, interpErr := kurtosis_types.SafeCastToString(codeVal, "exec code")
+	if interpErr != nil {
+		return interpErr
+	}
+	planYaml.storeFutureReference(uuid, codeFutureRef, "code")
+	outputVal, found, err := returnValue.Get(starlark.String("output"))
+	if err != nil {
+		return err
+	}
+	if !found {
+		return stacktrace.NewError("No code value found on exec dict")
+	}
+	outputFutureRef, interpErr := kurtosis_types.SafeCastToString(outputVal, "exec output")
+	if interpErr != nil {
+		return interpErr
+	}
+	planYaml.storeFutureReference(uuid, outputFutureRef, "output")
+
+	// create task yaml
+	taskYaml := &Task{} //nolint exhaustruct
+	taskYaml.Uuid = uuid
+	taskYaml.TaskType = EXEC
+	taskYaml.ServiceName = serviceName
+
+	cmdListWithFutureRefsSwapped := []string{}
+	for _, cmd := range cmdList {
+		cmdListWithFutureRefsSwapped = append(cmdListWithFutureRefsSwapped, planYaml.swapFutureReference(cmd))
+	}
+	taskYaml.RunCmd = cmdListWithFutureRefsSwapped
+	taskYaml.AcceptableCodes = acceptableCodes
+
+	planYaml.privatePlanYaml.Tasks = append(planYaml.privatePlanYaml.Tasks, taskYaml)
+	return nil
+}
+
+func (planYaml *PlanYaml) AddRenderTemplates(filesArtifactName string, filepaths []string) error {
+	uuid := planYaml.generateUuid()
+	filesArtifactYaml := &FilesArtifact{} //nolint exhaustruct
+	filesArtifactYaml.Uuid = uuid
+	filesArtifactYaml.Name = filesArtifactName
+	filesArtifactYaml.Files = filepaths
+	planYaml.addFilesArtifactYaml(filesArtifactYaml)
+	return nil
+}
+
+func (planYaml *PlanYaml) AddUploadFiles(filesArtifactName, locator string) error {
+	uuid := planYaml.generateUuid()
+	filesArtifactYaml := &FilesArtifact{} //nolint exhauststruct
+	filesArtifactYaml.Uuid = uuid
+	filesArtifactYaml.Name = filesArtifactName
+	filesArtifactYaml.Files = []string{locator}
+	planYaml.addFilesArtifactYaml(filesArtifactYaml)
+	return nil
+}
+
+func (planYaml *PlanYaml) AddStoreServiceFiles(filesArtifactName, locator string) error {
+	uuid := planYaml.generateUuid()
+	filesArtifactYaml := &FilesArtifact{} //nolint exhaustruct
+	filesArtifactYaml.Uuid = uuid
+	filesArtifactYaml.Name = filesArtifactName
+	filesArtifactYaml.Files = []string{locator}
+	planYaml.addFilesArtifactYaml(filesArtifactYaml)
+	return nil
+}
+
+func (planYaml *PlanYaml) RemoveService(serviceName string) {
+	for idx, service := range planYaml.privatePlanYaml.Services {
+		if service.Name == serviceName {
+			planYaml.privatePlanYaml.Services[idx] = planYaml.privatePlanYaml.Services[len(planYaml.privatePlanYaml.Services)-1]
```

Comment:
ah i see what you are doing

Reply:
we are on latest golang i wonder if you can use slices.Delete

Reply:
oh ye thats much better good callout